### PR TITLE
update message of training

### PR DESCRIPTION
**Pull request by VictorZhou123**
config pr: https://github.com/opensourceways/infra-mindspore/pull/1235

### CODE REVIEW

File: app/training.go [hunk 1]
```diff
@@ -127,10 +127,11 @@ func (s trainingService) create(
...
 
-	msg := new(message.MsgTraining)
-	msg.TrainingCreate(user, index, config.Inputs)
-
-	if err = s.sender.CreateTraining(msg); err != nil {
+	if err = s.sender.SendTrainingCreated(&domain.TrainingCreatedEvent{
+		Account:        user,
+		TrainingIndex:  index,
+		TrainingInputs: config.Inputs,
+	}); err != nil {
```
Comment (zengchen1024) [inline app/training.go:134]: 建议采用如下格式
err = s.sender.SendTrainingCreated(&domain.TrainingCreatedEvent{
		Account:        user,
		TrainingIndex:  index,
		TrainingInputs: config.Inputs,
	})
 if err != nil {

File: controller/messagequeue/training.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
+package messagequeue
```
Comment (zengchen1024) [inline controller/messagequeue/training.go:1]: 把 messagequeue 挪到跟controller平行的目录

File: controller/messagequeue/training.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+type TrainingConfig struct {
+	MaxRetry         int    `json:"max_retry"         required:"true"`
+	TrainingEndpoint string `json:"training_endpoint" required:"true"`
+
+	Topics TopicConfig `json:"topics"`
+}
+
+type TopicConfig struct {
+	TrainingCreated string `json:"training_created" required:"true"`
```
Comment (zengchen1024) [inline controller/messagequeue/training.go:123]: TrainingCreated 挪到 TrainingConfig。简单点

File: controller/messagequeue/training.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+		}
+	}
+
+	time.Sleep(interval)
+
+	return f(true)
+}
+
+type TrainingConfig struct {
+	MaxRetry         int    `json:"max_retry"         required:"true"`
```
Comment (zengchen1024) [inline controller/messagequeue/training.go:116]: MaxRetry 设置默认值

File: controller/messagequeue/training.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+				}
+			}
+
+			return err
+		},
+		10*time.Second,
+	)
+}
+
+func (c *consumer) retry(f func(bool) error, interval time.Duration) (err error) {
```
Comment (zengchen1024) [inline controller/messagequeue/training.go:95]: https://github.com/opensourceways/xihe-server/blob/dev/utils/util.go#L125 直接使用这个函数吧

File: controller/messagequeue/training.go [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+		return
+	}
+
+	v.Project.Id = b.Details["project_id"]
+	v.TrainingId = b.Details["training_id"]
+
+	return c.handleEventTrainCreated(&v)
+}
+
+func (c *consumer) handleEventTrainCreated(info *domain.TrainingIndex) error {
```
Comment (zengchen1024) [inline controller/messagequeue/training.go:68]: s/handleEventTrainCreated/createJob

File: message-server/config.go [hunk 7]
```diff
@@ -150,4 +151,7 @@ type mqTopics struct {
 
 	//course
 	CourseApplied string `json:"course_applied"     required:"true"`
+
+	// training
+	TrainingTopics msgqueue.TopicConfig `json:"training_topics"`
```
Comment (zengchen1024) [inline message-server/config.go:156]: TrainingCreated string `json:"training_created"`

File: message-server/main.go [hunk 8]
```diff
@@ -160,6 +169,27 @@ func bigmodelSubscribesMessage(cfg *configuration, topics *mqTopics) error {
 	)
 }
 
+func trainingSubscribeMessage(log *logrus.Entry, cfg *configuration) error {
```
Comment (zengchen1024) [inline message-server/main.go:172]: trainingSubscribesMessage

File: server/gin.go [hunk 9]
```diff
@@ -267,7 +270,7 @@ func setRouter(engine *gin.Engine, cfg *config.Config) {
 		)
 
 		controller.AddRouterForTrainingController(
-			v1, trainingAdapter, training, model, proj, dataset, sender,
+			v1, trainingAdapter, training, model, proj, dataset, trainingSender,
```
Comment (zengchen1024) [inline server/gin.go:273]: s/trainingSender/messages.NewTrainingMessageAdapter(&cfg.Training.Message, publisher) 记得换行

**Review by zengchen1024 — CHANGES_REQUESTED** [review]

File: app/training.go (hunk 1 above)
Comment (VictorZhou123) [reply app/training.go:134]: > 建议采用如下格式 err = s.sender.SendTrainingCreated(&domain.TrainingCreatedEvent{ Account: user, TrainingIndex: index, TrainingInputs: config.Inputs, }) if err != nil {

done

File: controller/messagequeue/training.go (hunk 2 above)
Comment (VictorZhou123) [reply controller/messagequeue/training.go:1]: > 把 messagequeue 挪到跟controller平行的目录

done

File: controller/messagequeue/training.go (hunk 3 above)
Comment (VictorZhou123) [reply controller/messagequeue/training.go:123]: > TrainingCreated 挪到 TrainingConfig。简单点

done

File: controller/messagequeue/training.go (hunk 5 above)
Comment (VictorZhou123) [reply controller/messagequeue/training.go:95]: > https://github.com/opensourceways/xihe-server/blob/dev/utils/util.go#L125 直接使用这个函数吧

done

File: controller/messagequeue/training.go (hunk 6 above)
Comment (VictorZhou123) [reply controller/messagequeue/training.go:68]: > s/handleEventTrainCreated/createJob

done

File: message-server/config.go (hunk 7 above)
Comment (VictorZhou123) [reply message-server/config.go:156]: > TrainingCreated string `json:"training_created"`

done

File: message-server/main.go (hunk 8 above)
Comment (VictorZhou123) [reply message-server/main.go:172]: > trainingSubscribesMessage

done

File: server/gin.go (hunk 9 above)
Comment (VictorZhou123) [reply server/gin.go:273]: > s/trainingSender/messages.NewTrainingMessageAdapter(&cfg.Training.Message, publisher) 记得换行

done

File: config/training.go [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+import (
+	"github.com/opensourceways/xihe-server/infrastructure/messages"
+	"github.com/opensourceways/xihe-server/infrastructure/trainingimpl"
+)
+
+type trainingConfig struct {
+	trainingimpl.Config
+
+	Message messages.TrainingConfig
```
Comment (zengchen1024) [inline config/training.go:11]: Message 缺少tag

File: messagequeue/training.go [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+		return
+	}
+
+	v.Project.Id = b.Details["project_id"]
+	v.TrainingId = b.Details["training_id"]
+
+	return c.handleEventCreateJob(&v)
+}
+
+func (c *consumer) handleEventCreateJob(info *domain.TrainingIndex) error {
```
Comment (zengchen1024) [inline messagequeue/training.go:69]: s/handleEventCreateJob/createJob

File: messagequeue/training.go [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+	// wait for the sync of model and dataset
+	time.Sleep(10 * time.Second)
+
+	utils.RetryThreeTimes(func() error {
+		retry, err := c.s.CreateTrainingJob(
+			info, c.cfg.TrainingEndpoint, true,
+		)
+		if err != nil {
+			c.log.Errorf(
```
Comment (zengchen1024) [inline messagequeue/training.go:78]: logrus.Errorf

File: message-server/config.go [hunk 13]
```diff
@@ -150,4 +150,7 @@ type mqTopics struct {
 
 	//course
 	CourseApplied string `json:"course_applied"     required:"true"`
+
```
Comment (zengchen1024) [inline message-server/config.go:153]: 去掉 34， 35行，用messagequeue中的config作为training的config

**Review by zengchen1024 — CHANGES_REQUESTED** [review]

File: server/gin.go [hunk 14]
```diff
@@ -270,7 +269,9 @@ func setRouter(engine *gin.Engine, cfg *config.Config) {
 		)
 
 		controller.AddRouterForTrainingController(
-			v1, trainingAdapter, training, model, proj, dataset, trainingSender,
+			v1, trainingAdapter, training, model, proj, dataset, messages.NewTrainingMessageAdapter(
+				&cfg.Training.Message, publisher,
```
[the remark this answers is not in the document]
Comment (VictorZhou123) [reply server/gin.go:273]: > messages.NewTrainingMessageAdapter 放到273行

done

File: config/training.go (hunk 10 above)
Comment (VictorZhou123) [reply config/training.go:11]: > Message 缺少tag

done

File: messagequeue/training.go (hunk 11 above)
Comment (VictorZhou123) [reply messagequeue/training.go:69]: > createJob

done

File: messagequeue/training.go (hunk 12 above)
Comment (VictorZhou123) [reply messagequeue/training.go:78]: > logrus.Errorf

done

File: message-server/config.go (hunk 13 above)
Comment (VictorZhou123) [reply message-server/config.go:153]: > 去掉 34， 35行，用messagequeue中的config作为training的config

done

File: messagequeue/training.go [hunk 15]
```diff
@@ -62,63 +63,45 @@ func (c *consumer) handleEventTrainingCreated(body []byte, h map[string]string)
...
 	TrainingEndpoint string `json:"training_endpoint" required:"true"`
 
-	Topics TopicConfig `json:"topics"`
+	// topic
+	TrainingCreated string `json:"training_created" required:"true"`
 }
 
-type TopicConfig struct {
-	TrainingCreated string `json:"training_created" required:"true"`
+func (cfg *TrainingConfig) SetDefault() {
```
[the remark this answers is not in the document]
Comment (VictorZhou123) [reply messagequeue/training.go:103]: > TrainingCreated 应该作为参数传入

done

File: message-server/config.go [hunk 16]
```diff
@@ -70,8 +70,8 @@ func (cfg *configuration) ConfigItems() []interface{} {
 }
 
 func (cfg *configuration) setDefault() {
-	if cfg.MaxRetry <= 0 {
-		cfg.MaxRetry = 10
+	if cfg.Training.MaxRetry <= 0 {
```
Comment (zengchen1024) [inline message-server/config.go:73]: 删除73， 74

File: server/gin.go [hunk 17]
```diff
@@ -269,9 +266,9 @@ func setRouter(engine *gin.Engine, cfg *config.Config) {
 		)
 
 		controller.AddRouterForTrainingController(
-			v1, trainingAdapter, training, model, proj, dataset, messages.NewTrainingMessageAdapter(
-				&cfg.Training.Message, publisher,
-			),
+			v1, trainingAdapter, training, model, proj, dataset,
+			messages.NewTrainingMessageAdapter(
+				&cfg.Training.Message, publisher),
```
Comment (zengchen1024) [inline server/gin.go:271]: messages.NewTrainingMessageAdapter(
				&cfg.Training.Message, publisher，
),

File: message-server/main.go [hunk 18]
```diff
@@ -197,8 +192,8 @@ func newHandler(cfg *configuration, log *logrus.Entry) *handler {
 
 	h := &handler{
 		log:              log,
-		maxRetry:         cfg.MaxRetry,
-		trainingEndpoint: cfg.TrainingEndpoint,
+		maxRetry:         cfg.Training.MaxRetry,
+		trainingEndpoint: cfg.Training.TrainingEndpoint,
```
Comment (zengchen1024) [inline message-server/main.go:196]: 为什么还要配置trainingEndpoint

File: server/gin.go [hunk 19]
```diff
@@ -94,9 +94,6 @@ func setRouter(engine *gin.Engine, cfg *config.Config) {
 		mongodb.NewDatasetMapper(collections.Dataset),
 	)
 
-	// user := repositories.NewUserRepository(
```
Comment (zengchen1024) [inline server/gin.go:97]: 删除无效的配置

**Review by zengchen1024 — CHANGES_REQUESTED** [review]

File: message-server/config.go (hunk 16 above)
Comment (VictorZhou123) [reply message-server/config.go:73]: > 删除73， 74

done

File: message-server/main.go (hunk 18 above)
Comment (VictorZhou123) [reply message-server/main.go:196]: > 为什么还要配置trainingEndpoint

done

File: message-server/config.go [hunk 20]
```diff
@@ -70,8 +71,8 @@ func (cfg *configuration) ConfigItems() []interface{} {
 }
 
 func (cfg *configuration) setDefault() {
-	if cfg.MaxRetry <= 0 {
-		cfg.MaxRetry = 10
+	if cfg.MaxRetry == 0 {
```
Comment (zengchen1024) [inline message-server/config.go:74]: <=

**Review by zengchen1024 — CHANGES_REQUESTED** [review]

**Pull request merged**